src/compiler/types.ts

```diff
@@ -2870,6 +2870,7 @@ namespace ts {
     /* @internal */
     export interface TransientSymbol extends Symbol, SymbolLinks {
         checkFlags: CheckFlags;
+        isRestParameter?: boolean; 

```

Comment:
No, we create symbols for all kinds of reasons. This is cheaper than creating an entire synthetic node just to trick `buildParameterDisplay`. At one point I was looking into creating purely synthetic types, and this was one of the required changes to make this work.